include/exadg/solvers_and_preconditioners/multigrid/multigrid_parameters.h

```diff
@@ -104,6 +110,7 @@ struct AMGData
     ml_data.smoother_sweeps = 1;
     ml_data.n_cycles        = 1;
     ml_data.smoother_type   = "ILU";
+    ml_operator_type        = MLOperatorType::Laplace;
```

Comment:
Related to comment https://github.com/exadg/exadg/pull/693#issuecomment-2315343733, it is in my opinion hard to understand for a user what the code actually does internally. Do I have to set this parameter in an application-file or not? The declaration of the type `MLOperatorType` does currently not provide documentation. My opinion is currently that this is an internal parameter that ExaDG can deduce from the PDE model problem (Poisson, Structure) that is considered.

If not realized as an internal parameter (but as a user parameter as done currently), we typically use the enum value `Undefined` so that the code is more transparent to the user. When not setting the parameter and using the default value `Undefined`, the user gets an error message stating that the parameter has not been defined so far, ideally only for those cases  where this parameter is going to be used by the code. I like this style because this means for a user that one does not have to study all the internal implementations in ExaDG in order to be sure that the code really does what one expects it to do. I understand that the present implementation requires minimal changes to the code and in the sense that a user only has to set this parameter for Elasticity problems and do/change nothing for e.g. Poisson. At the same time I have concerns whether we really benefit from this style because of the implicit knowledge/information involved.

Reply:
I would like to go for: introduce `Undefined` to print it to screen (we can also skip this, I have no preference, but for safety in the tests it would be good to make sure it is set correctly).
And then later set it during setup.

Reply:
I would rather set it to `Laplace` by default (internal variable), where `Elasticity` is then the exception.